src/main/scala/ai/starlake/job/sink/bigquery/BigQuerySparkJob.scala

```diff
@@ -303,14 +305,21 @@ object BigQuerySparkJob {
 
   def getTable(
     session: SparkSession,
-    datasetName: String,
-    tableName: String
+    resourceId: String
   ): TableMetadata = {
-    val conf = session.sparkContext.hadoopConfiguration
-    val projectId: String =
-      Option(conf.get("fs.gs.project.id")).getOrElse(ServiceOptions.getDefaultProjectId)
-    val bigquery: BigQuery = BigQueryOptions.getDefaultInstance().getService()
-    val tableId = TableId.of(projectId, datasetName, tableName)
-    TableMetadata(Option(bigquery.getTable(tableId)), bigquery)
+    val finalTableId = getTableId(session, resourceId)
+    val bigquery = BigQueryOptions.getDefaultInstance().getService()
```

Comment:
This should never be used.
We should call bigquery() that check if a service account has been passed by the user for execution.

This has been added to support SaaS mode where Starlaek will serve multiple requests coming from different users.
This is also why we do not have the object BigQueryJobBase anymore and everything has been moved to the trait.